client/src/components/AddCreditCard/AddCreditCard.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+import { CardElement, useStripe, useElements } from '@stripe/react-stripe-js';
+import axios from 'axios';
+import useStyles from './useStyles';
+import { Button } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Snackbar from '@material-ui/core/Snackbar';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+interface IProps {
+  saveCard?: boolean;
+  price?: number;
+  closeModel?: () => void;
+}
+export const AddCreditCard = (Props: IProps): JSX.Element => {
+  const stripe = useStripe();
+  const elements = useElements();
+  const classes = useStyles();
+  const [isLoading, setLoading] = useState(false);
+  const [open, setOpen] = React.useState(false);
+  const [message, setMessage] = useState('');
+  const { saveCard = false, price, closeModel } = Props;
+
+  const handleClose = () => {
+    if (closeModel) {
+      closeModel();
+    } else {
+      setOpen(false);
+    }
+  };
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      setLoading(true);
+      return;
+    }
+
+    const element = elements.getElement(CardElement);
+
+    if (element === null) {
+      setLoading(true);
+      return;
+    }
+
+    const { error, paymentMethod } = await stripe.createPaymentMethod({
+      type: 'card',
+      card: element,
+    });
+
+    if (!error) {
```

Comment:
You could save yourself the double conditional block by doing an early return if there's an error (as above)